purchases/src/main/kotlin/com/revenuecat/purchases/CustomerInfoRetriever.kt

```diff
@@ -0,0 +1,141 @@
+package com.revenuecat.purchases
+
+import android.os.Handler
+import android.os.Looper
+import android.util.Log
+import com.revenuecat.purchases.common.Backend
+import com.revenuecat.purchases.common.LogIntent
+import com.revenuecat.purchases.common.caching.DeviceCache
+import com.revenuecat.purchases.common.debugLog
+import com.revenuecat.purchases.common.errorLog
+import com.revenuecat.purchases.common.log
+import com.revenuecat.purchases.interfaces.ReceiveCustomerInfoCallback
+import com.revenuecat.purchases.strings.CustomerInfoStrings
+
+internal class CustomerInfoRetriever(
+    private val deviceCache: DeviceCache,
+    private val backend: Backend,
+    private val handler: Handler = Handler(Looper.getMainLooper())
+) {
+
+    fun retrieveCustomerInfo(
+        appUserID: String,
+        fetchPolicy: CacheFetchPolicy,
+        appInBackground: Boolean,
+        cacheCustomerInfoAfterFetchCallback: (CustomerInfo) -> Unit,
+        callback: ReceiveCustomerInfoCallback? = null
+    ) {
+        debugLog(CustomerInfoStrings.RETRIEVING_CUSTOMER_INFO.format(fetchPolicy))
+        when (fetchPolicy) {
+            CacheFetchPolicy.CACHE_ONLY -> getCustomerInfoCacheOnly(appUserID, callback)
+            CacheFetchPolicy.FETCH_CURRENT -> getCustomerInfoFetchOnly(
+                appUserID,
+                appInBackground,
+                cacheCustomerInfoAfterFetchCallback,
+                callback
+            )
+            CacheFetchPolicy.CACHED_OR_FETCHED -> getCustomerInfoCachedOrFetched(
+                appUserID,
+                appInBackground,
+                cacheCustomerInfoAfterFetchCallback,
+                callback
+            )
+            CacheFetchPolicy.NOT_STALE_CACHED_OR_CURRENT -> getCustomerInfoNotStaledCachedOrFetched(
+                appUserID,
+                appInBackground,
+                cacheCustomerInfoAfterFetchCallback,
+                callback
+            )
+        }
+    }
+
+    private fun getCustomerInfoCacheOnly(
+        appUserID: String,
+        callback: ReceiveCustomerInfoCallback? = null
+    ) {
+        if (callback == null) return
```

Comment:
I think this is ok... Since we basically won't be getting any value back, but lmk if you have any concerns.